General/TweakingBigImages.md

```diff
@@ -0,0 +1,116 @@
+# Tweaking big images
+
+Pharo is special in the sense that the language and IDE are mixed. This mean that your application runs into the IDE.
+In some cases we might end-up Pharo images having a large memory consumption (aka Big Images). 
+The goal of this page is to sum up some tweaking that can be done on Pharo images to improve the performances on such images. 
+
+* [Fast dragging](#fast-dragging)
+* [Disable taskbar previews](#disable-taskbar-previews)
+* [Tune Garbage Collection](#tune-garbage-collection)
+* [Disable Quality Assistant plugin](#disable-quality-assistant-plugin)
+
+## Fast dragging
+
+While dragging windows in Pharo an animation happens. This animation is not performant and can make the dragging of windows slow in large images.
+
+It is possible to workaround this problem by enabling a "fast dragging" option:
+
+```Smalltalk
+UITheme currentSettings fastDragging: true
+```
+
+This option is reset when we change the theme Pharo and should thus be executed each time you update the theme.
+
+## Disable taskbar previews
+
+While going oven the taskbar of Pharo, previews of the images are displayed. This feature can be really slow on big images. 
+It is possible to disable it:
+
+```Smalltalk
+TaskbarMorph showWindowPreview: false
+```
+
+## Tune Garbage Collection 
+
+Pharo is a language with a memory management using garbage collection. 
+When objects are created, memory is allocated to them. Once in a while, the memory is cleaned and the "garbage collector" free the memory of objects that are not been used anymore.
+
+This process can be customized via multiple parameters. To understand the impact of those parameters we explain here how the garbage collection works.
+
+Pharo memory (aka heap) is divided into multiple spaces:
+- The "Old space" contains objects that survived multiples garbage collections (GC). When it is filled, new memory segments are allocated to the old space.
+- The "New Space" contains recenttly created objects. 
+
+The "New Space" is divided into 3 sub spaces:
+- The "Eden" containing the newest objects created (taking 5/7 of the new space)
+- Two survivors spaces (1/7 of the new space each) contains the object that survived some garbage collection already
+
+The Garbage Collector execute two kind of cleanings:
+- The scavenge happens after the Eden is filled at a certain ratio and will clean the Eden. When objects from the eden are still referenced, they are moved to the survivors. There are two survivors: past survivor space and future survivor space (past space and future space for short). Past and future space swap after each scavenge, which empties eden and past space, copying survivors into future space, tenuring overflow to old space and then making future space the new past space. When the survivor space is fill at more than 90%, the survivors are moved to the old space.
+- The full GC happens when the heap grow past a certain ratio. It will go through the old space and clean the old objects.
```

Comment:
```suggestion
- The **full garbage collect** (aka full GC) happens when the heap grow past a certain ratio. It will go through the old space and clean the old objects.
```